File: mediq-frontend/src/services/userService.js
import api from './api';

export const userService = {
  createUser: async (data) => api.post('/users', data),
  updateUser: async (id, data) => api.put(`/users/${id}`, data),
  getUserById: async (id) => api.get(`/users/${id}`),
  getUserByMemberId: async (memberId) => api.get(`/users/member/${memberId}`),
  getUsersByRole: async (role) => api.get(`/users/role/${role}`),
  searchUsers: async (role, keyword, page = 0, size = 10) =>
    api.get('/users', { params: { role, keyword, page, size } }),
  activateUser: async (id) => api.patch(`/users/${id}/activate`),
  deactivateUser: async (id) => api.patch(`/users/${id}/deactivate`),
  deleteUser: async (id) => api.delete(`/users/${id}`),
  resetPassword: async (id, newPassword) => api.patch(`/users/${id}/reset-password`, { newPassword }),
};

export default userService;

File: mediq-backend/src/main/java/com/mediq/service/UserService.java
package com.mediq.service;

import com.mediq.constants.UserRole;
import com.mediq.dto.CreateUserRequest;
import com.mediq.dto.ResetPasswordRequest;
import com.mediq.dto.UpdateUserRequest;
import com.mediq.dto.UserResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public interface UserService {

    UserResponse createUser(CreateUserRequest request);

    UserResponse updateUser(Long userId, UpdateUserRequest request);

    UserResponse getUserById(Long userId);

    UserResponse getUserByMemberId(String memberId);

    List<UserResponse> getUsersByRole(UserRole role);

    Page<UserResponse> searchUsers(UserRole role, String keyword, int page, int size, String sortBy, String sortDir);

    UserResponse activateUser(Long userId);

    UserResponse deactivateUser(Long userId);

    void deleteUser(Long userId);

    void resetPassword(Long userId, ResetPasswordRequest request);
}
